src/main/java/servlets/PlacesListServlet.java

```diff
@@ -33,12 +33,14 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
         Double lat = Double.parseDouble(request.getParameter("lat"));
         Double lng = Double.parseDouble(request.getParameter("lng"));
         LatLng currentLocation = new LatLng(lat, lng);
+        String filters = request.getParameter("filter").replace(',', '|');
```

Comment:
What is the purpose of replace here?

Reply:
Arrays are given by comma-separated values in html queries, however places API was using '|' to show that it is a multiple-keyword 'keyword', if it make sense. They have replied saying that we can try using '+' and it should produce better results.

I have tested with '+' and it indeed returns more consistent results. Patchset 5855d1a implements the change